docs/examples/algo/complete-example.py

```diff
@@ -1,144 +1,171 @@
 import csv
 import datetime
-
+import yaml
 import numpy
 
 from nupic.encoders.date import DateEncoder
 from nupic.encoders.random_distributed_scalar import \
-    RandomDistributedScalarEncoder
+  RandomDistributedScalarEncoder
 from nupic.research.spatial_pooler import SpatialPooler
 from nupic.research.temporal_memory import TemporalMemory
 from nupic.algorithms.sdr_classifier_factory import SDRClassifierFactory
 
-_INPUT_FILE_PATH = "../data/gymdata.csv"
+_NUM_RECORDS = 3000
+_INPUT_FILE_PATH = '../data/gymdata.csv'
+_OUTPUT_FILE_PATH = 'predictions.csv'
+_PARAMS_PATH = '../params/model.yaml'
 
-def runHotgym():
 
-  timeOfDayEncoder = DateEncoder(timeOfDay=(21,1))
-  weekendEncoder = DateEncoder(weekend=21)
-  scalarEncoder = RandomDistributedScalarEncoder(0.88)
 
-  encodingWidth = timeOfDayEncoder.getWidth() \
-    + weekendEncoder.getWidth() \
-    + scalarEncoder.getWidth()
+def runHotgym():
+  with open(_PARAMS_PATH, 'r') as f:
+    modelParams = yaml.safe_load(f)['modelParams']
+    enParams = modelParams['sensorParams']['encoders']
+    spParams = modelParams['spParams']
+    tmParams = modelParams['tmParams']
+    clParams = modelParams['clParams']
+
+  timeOfDayEncoder = DateEncoder(
+    timeOfDay=enParams['timestamp_timeOfDay']['timeOfDay'])
+  weekendEncoder = DateEncoder(
+    weekend=enParams['timestamp_weekend']['weekend'])
+  scalarEncoder = RandomDistributedScalarEncoder(
+    enParams['consumption']['resolution'])
+
+  encodingWidth = (timeOfDayEncoder.getWidth()
+                   + weekendEncoder.getWidth()
+                   + scalarEncoder.getWidth())
 
   sp = SpatialPooler(
     # How large the input encoding will be.
     inputDimensions=(encodingWidth),
     # How many mini-columns will be in the Spatial Pooler.
-    columnDimensions=(2048),
+    columnDimensions=(spParams['columnCount']),
```

Comment:
This is @rhyolight 's example. I think he pulled the params one by one so that he can describe them in the the quick-start guide: http://nupic.docs.numenta.org/quick-start/algorithms.html